src/display/canvas.js

```diff
@@ -1215,6 +1156,19 @@ class CanvasGraphics {
       cache.clear();
     }
     this._cachedBitmapsMap.clear();
+
+    if (applyHCMFilter && this._pageColors) {
+      const hcmFilterId = this.filterFactory.computeHCMFilter(
+        this._pageColors.foreground,
+        this._pageColors.background
+      );
+      if (hcmFilterId) {
```

Comment:
For consistency with the pre-existing `BaseFilterFactory` method, how about having this new method returning `"none"` (instead of an empty string) and check for that here?